tests/test_views.py

```diff
@@ -1503,6 +1573,121 @@ def test_get_course_enrollment_page_with_unparseable_course_effort(
             'duration': '12 weeks per course',
             'course_level_type': 'Type 1',
             'close_modal_button_text': 'Close',
+            'premium_modes': premium_modes,
+        }
+        for key, value in expected_context.items():
+            assert response.context[key] == value  # pylint: disable=no-member
+
+    @mock.patch('enterprise.views.render', side_effect=fake_render)
+    @mock.patch('enterprise.views.configuration_helpers')
+    @mock.patch('enterprise.tpa_pipeline.get_enterprise_customer_for_request')
+    @mock.patch('enterprise.views.get_real_social_auth_object')
+    @mock.patch('enterprise.views.quarantine_session')
+    @mock.patch('enterprise.views.lift_quarantine')
+    @mock.patch('enterprise.views.CourseApiClient')
+    @mock.patch('enterprise.views.organizations_helpers')
+    @mock.patch('enterprise.views.CourseCatalogApiClient')
+    @mock.patch('enterprise.views.ecommerce_api_client')
+    def test_get_course_enrollment_page_with_ecommerce_error(
```

Comment:
@replaceafill This test has a lot of logic that overlaps with the logic from the tests that got added in #110. Once you've refactored those, see if you can reuse existing logic here (or use `ddt`) to reduce duplication.